src/v/cluster/controller.cc

```diff
@@ -446,39 +467,113 @@ ss::future<> controller::stop() {
     });
 }
 
+ss::future<> controller::create_cluster() {
+    vassert(
+      ss::this_shard_id() == controller_stm_shard,
+      "Cluster can only be created from controller_stm_shard");
+    bootstrap_cluster_cmd_data cmd_data;
+    cmd_data.uuid = model::cluster_uuid(uuid_t::create());
+    cmd_data.bootstrap_user_cred
+      = security_frontend::get_bootstrap_user_creds_from_env();
+    vlog(clusterlog.info, "Creating cluster {}", cmd_data.uuid);
+
+    for (simple_time_jitter<model::timeout_clock> retry_jitter(1s);;) {
+        model::record_batch b = serde_serialize_cmd(
+          bootstrap_cluster_cmd{0, cmd_data});
+        // cluster::replicate_and_wait() cannot be used here because it checks
+        // for serde_raft_0 feature to be enabled. Before a cluster is here,
+        // no feature is active, however the bootstrap_cluster_cmd is serde-only
```

Comment:
nit: update comment

Reply:
It is up to date, `cluster::replicate_and_wait()` is avoided here for exactly that reason.

Reply:
Ah oops, missed that this was the `cluster::` variant.